features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,582 @@
+  } = Collateral.useContainer();
+  const { activeSponsors } = EmpSponsors.useContainer();
+
+  const [dialogTabIndex, setDialogTabIndex] = useState<string | null>(
+    "deposit"
+  );
+  const [collateralToDeposit, setCollateralToDeposit] = useState<string>("");
+  const [minCollPerToken, setMinCollPerToken] = useState<string>("");
+  const [maxCollPerToken, setMaxCollPerToken] = useState<string>("");
+  const [maxTokensToLiquidate, setMaxTokensToLiquidate] = useState<string>("");
+  const [deadline, setDeadline] = useState<string>("");
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const {
+    priceIdentifier: priceId,
+    collateralRequirement: collReq,
+    currentTime,
+  } = empState;
+
+  const setDialogTab = (
+    event: MouseEvent<HTMLElement>,
+    newAlignment: string | null
+  ) => {
+    setDialogTabIndex(newAlignment);
+  };
+
+  const prettyBalance = (x: number) => {
+    const x_string = x.toFixed(4);
+    return utils.commify(x_string);
+  };
+
+  const prettyAddress = (x: String | null) => {
+    if (!x) return "N/A";
+    return x.substr(0, 6) + "..." + x.substr(x.length - 6, x.length);
+  };
+  if (
+    props.selectedSponsor &&
+    activeSponsors[props.selectedSponsor] &&
+    latestPrice &&
+    collReq
+  ) {
+    const sponsorPosition = activeSponsors[props.selectedSponsor];
+
+    const pendingWithdrawTimeRemaining =
+      Number(sponsorPosition.withdrawalTimestamp) - Number(currentTime);
+
+    const pendingWithdrawTimeString =
+      pendingWithdrawTimeRemaining && pendingWithdrawTimeRemaining > 0
+        ? Math.max(0, Math.floor(pendingWithdrawTimeRemaining / 3600)) +
+          ":" +
+          Math.max(0, Math.floor((pendingWithdrawTimeRemaining % 3600) / 60)) +
+          ":" +
+          Math.max(0, (pendingWithdrawTimeRemaining % 3600) % 60)
+        : "None";
+
+    const pendingTransferTimeRemaining =
+      Number(sponsorPosition.TransferTimestamp) - Number(currentTime);
+
+    const pendingTransferTimeString =
+      pendingTransferTimeRemaining && pendingTransferTimeRemaining > 0
+        ? Math.max(0, Math.floor(pendingTransferTimeRemaining / 3600)) +
+          ":" +
+          Math.max(0, Math.floor((pendingTransferTimeRemaining % 3600) / 60)) +
+          ":" +
+          Math.max(0, (pendingTransferTimeRemaining % 3600) % 60)
+        : "None";
+
+    const underCollateralizedPrice =
+      Number(sponsorPosition.collateral) /
+      (Number(sponsorPosition.tokensOutstanding) *
+        parseFloat(fromWei(collReq)));
+
+    const underCollateralizedPercent =
+      ((Number(latestPrice) - Number(underCollateralizedPrice)) /
+        Number(latestPrice)) *
+      100;
+
+    const collBalanceTooLow = () => {
+      if (dialogTabIndex == "deposit") {
+        return (collBalance || 0) < (Number(collateralToDeposit) || 0);
```

Comment:
I think its more readable if you declare `collBalanceNum = collBalance || 0` at the outset of the file since you do use this logic elsewhere. Same with `collateralToDeposit` and `finalFee` etc.

Reply:
sure. that's more consistent with how we do it elsewhere. I've changed this, PTAL.